packages/glimmer-runtime/tests/ember-component-test.ts

```diff
@@ -578,37 +547,78 @@ testComponent('parameterized hasBlock (inverse) when inverse supplied', {
   expected: 'Yes'
 });
 
-testComponent('parameterized hasBlock (inverse) when inverse not supplied', {
-  skip: true,
-  layout: '{{#if (hasBlock "inverse")}}Yes{{else}}No{{/if}}',
+testComponent('parameterized has-block (subexpr, inverse) when inverse not supplied', {
+  layout: '{{#if (has-block "inverse")}}Yes{{else}}No{{/if}}',
   invokeAs: { template: 'block here' },
   expected: 'No'
 });
 
-testComponent('parameterized hasBlock (default) when block supplied', {
-  skip: true,
-  layout: '{{#if (hasBlock)}}Yes{{else}}No{{/if}}',
+testComponent('parameterized has-block (subexpr, default) when block supplied', {
+  layout: '{{#if (has-block)}}Yes{{else}}No{{/if}}',
   invokeAs: { template: 'block here' },
   expected: 'Yes'
 });
 
-testComponent('parameterized hasBlock (default) when block not supplied', {
-  skip: true,
-  layout: '{{#if (hasBlock)}}Yes{{else}}No{{/if}}',
+testComponent('parameterized has-block (subexpr, default) when block not supplied', {
+  kind: 'curly',
+  layout: '{{#if (has-block)}}Yes{{else}}No{{/if}}',
   expected: 'No'
 });
 
-testComponent('hasBlock keyword when block supplied', {
-  skip: true,
-  layout: '{{#if hasBlock}}Yes{{else}}No{{/if}}',
+testComponent('parameterized has-block (content, inverse) when inverse supplied', {
+  kind: 'curly',
+  layout: '{{has-block "inverse"}}',
+  invokeAs: {
+    template: 'block here',
+    inverse: 'inverse here'
+  },
+  expected: 'true'
+});
+
+testComponent('parameterized has-block (content, inverse) when inverse not supplied', {
+  layout: '{{has-block "inverse"}}',
   invokeAs: { template: 'block here' },
-  expected: 'Yes'
+  expected: 'false'
 });
 
-testComponent('hasBlock keyword when block not supplied', {
-  skip: true,
-  layout: '{{#if hasBlock}}Yes{{else}}No{{/if}}',
-  expected: 'No'
+testComponent('parameterized has-block (content, default) when block supplied', {
+  layout: '{{has-block}}',
+  invokeAs: { template: 'block here' },
+  expected: 'true'
+});
+
+testComponent('parameterized has-block (content, default) when block not supplied', {
+  kind: 'curly',
+  layout: '{{has-block}}',
+  expected: 'false'
+});
+
+testComponent('parameterized has-block (attr, inverse) when inverse supplied', {
+  kind: 'curly',
+  layout: '<button class={{has-block "inverse"}}></button>',
```

Comment:
Can you use some other test than `class={{unquoted}}`? The class prop/attr is the most entangled in likely changes.

Maybe better would be: `<button data-has-block="{{has-block "inverse"}}"></button>